#!/usr/bin/env python
PACKAGE = "duo_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Camera related variables
gen.add("gain",    double_t,    0, "Camera gain", 0,  0, 100)
gen.add("exposure",    double_t,    0, "Camera exposure (%)", 20,  0, 100)
gen.add("leds",    double_t,    0, "LED exposure (%)", 20,  0, 100)
#gen.add("ledSequence",    int_t,    0, "LED sequence step count", 0,  0, 64)
gen.add("flipH",	bool_t,    0, "Enable horizontal flip", False)
gen.add("flipV",	bool_t,    0, "Enable vertical flip", False)
gen.add("cameraSwap",	bool_t,    0, "Swaps left and right streams", False)
#gen.add("fps",       int_t,  0, "Frames per second captured",	30, 0, 100)
#gen.add("width",       int_t,  0, "Frame width (pixels)",	752, 0, 752)
#gen.add("height",       int_t,  0, "Frame height (pixels)",	480, 0, 480)
# Dense3D related variables
#gen.add("speckleWindowSize",       int_t,  0, "Speckle Window Size (dense3D)",	0, 0, 256)
#gen.add("speckleRange",   int_t,  0, "Speckle Range (dense3D)",	0, 0, 16)
#gen.add("uniquenessRatio",	      int_t,  0, "Uniqueness Ratio (dense3D)",	0, 0, 100)
#gen.add("preFilterCap",   int_t,  0, "Pre-filter cap (dense3D)",	0,0, 256)
#gen.add("sadWindowSize",   int_t,  0, "Pre-filter cap (dense3D)",	2,2, 10)
#gen.add("numDisparities",   int_t,  0, "Number of disparities (dense3D)",	3,2, 16)
#gen.add("dense3dMode",   int_t,  0, "Mode (0 - BM)", 0,0,3)

exit(gen.generate(PACKAGE, "duo_ros", "Duo"))